chainercv/datasets/ade20k/ade20k_semantic_segmentation_dataset.py

```diff
@@ -0,0 +1,103 @@
+import glob
+from multiprocessing import Pool
+import os
+
+import numpy as np
+
+from chainer import dataset
+from chainer.dataset import download
+from chainercv import utils
+from chainercv.utils import read_image
+
+root = 'pfnet/chainercv/ade20k'
+trainval_url = 'http://data.csail.mit.edu/places/ADEchallenge/'
+trainval_url += 'ADEChallengeData2016.zip'
+test_url = 'http://data.csail.mit.edu/places/ADEchallenge/release_test.zip'
+
+
+def get_ade20k():
+    p = Pool(2)
+    data_root = download.get_dataset_directory(root)
+    urls = [trainval_url, test_url]
+    ret = [p.apply_async(utils.cached_download, args=(url,)) for url in urls]
+    caches = [r.get() for r in ret]
+    args = [(cache_fn, data_root, os.path.splitext(url)[1])
+            for cache_fn, url in zip(caches, urls)]
+    ret = [p.apply_async(utils.extractall, args=arg) for arg in args]
+    for r in ret:
+        r.get()
+    return data_root
+
+
+class ADE20KSemanticSegmentationDataset(dataset.DatasetMixin):
+
+    """Semantic segmentation dataset for `ADE20K`_.
+
+    This is ADE20K dataset distributed in MIT Scene Parsing Benchmark website.
+    It has 20,210 training images, 2,000 validation images, and 3,352 test
+    images.
+
+    .. _`MIT Scene Parsing Benchmark`: http://sceneparsing.csail.mit.edu/
+
+    Args:
+        data_dir (string): Path to the dataset directory. The directory should
+            contain at least two directories, :obj:`annotations` and
+            :obj:`images`. If :obj:`auto` is given, the dataset is
+            automatically downloaded into
+            :obj:`$CHAINER_DATASET_ROOT/pfnet/chainercv/ade20k`.
+        split ({'train', 'val', 'test'}): Select from dataset splits used in
+            Cityscapes dataset.
```

Comment:
`Cityscapes`?

Reply:
Oops, I forgot to fix the name... Thanks for pointing out it!